### DISCUSSION TITLE
REST Api convert array to multi-select

### DISCUSSION THREAD

**Discussion by pausag** [Budibase/budibase#5762] [Help]
Hi, all! 
I have a question regarding transforming the data coming from REST API source. I have a json schema which looks like this: 
` {
  "nickname": "nickname 1",
  "chat": "chat 1",
  "help_requests": [
   "1",
   "2",
   "3",
   "4"
  ],
  "help_offers": [
   "1",
   "2",
   "3",
   "4"
  ],
  "active": false,
  "emergency": false,
  "supervisor_id": "supervisor_id 1",
  "id": "1"
 },`

It renders in a table row nicely, and CRUD works for everything, except the `help_request` field. The table row render: 

![image](https://user-images.githubusercontent.com/8569827/167253749-52b2fecf-3003-42a4-abc8-1cebf5fce6c6.png)

Is it possible to convert types of   `HELP_REQUESTS` and `HELP_OFFERS` to `multi-select`? 

it also applies to the modification option: I can only use `textField` to display the data, which looks like this: 

![image](https://user-images.githubusercontent.com/8569827/167253894-43596149-cd11-4f89-9fde-b66dff2211e9.png)

Is there any possibility to convert this field to multi-select? 

I've tried to wrap the REST data provider with another one, which is  connected to an internal table and defines data for multi-select field. This somehow works to create a new entry, but I could not display the existing values together with the data defined for the multi-select. 

Any help here would be much appreciated.

**Comment by mjashanks** [comment]
Hey Pavel.

Unfortunately, I wasn't able to build this. I almost got there, but found a bug on the way.

For your info (and anyone else), I intended to achieve it with the following steps
1. Add a multi-picker, and manually name the `Field` setting as `values_array`
2. Set the `Default Value` setting, using javascript to:

```javascript
const valuesInArray = `[${$("New Repeater.Get Something.values")}]`;
return JSON.parse(valuesInArray);
```
This works - i.e. it populates the picker correctly. However, when I use the app, I am unable to change the values in the picker. This is a bug, which I have logged: https://github.com/Budibase/budibase/issues/5765

Sorry I couldn't be of more help !

**Reply by pausag** [reply]
Thank you very much for your reply! Indeed, this works, except for selecting elements in the picker. For now, I just went with plain array of checkbox fields :) I know it is not nice, but it works for now. Will wait for the version update. Thank you!
